sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -538,6 +539,13 @@ def _insert_all_rows(
     try:
       response = self.client.tabledata.InsertAll(request)
       # response.insertErrors is not [] if errors encountered.
+    except HttpError as exn:
+      if error_counter:
+        content = json.loads(exn.content)
+        error_counter.record(
+            '%s(%s)' %
+            (content['error']['errors'][0]['reason'], exn.status_code))
```

Comment:
Please convert this to the human readable format instead of the numeric http code. See the doc I shared with you.

At least before logging
https://cloud.google.com/apis/design/errors

Reply:
It records string error codes and numeric http codes from BigQuery API JSON response https://cloud.google.com/bigquery/docs/error-messages. For example:
```
{quotaExceeded(403): 461, rateLimitExceeded(403): 2724}
```
I think string error codes here are human readable and clearly shows the cause of errors. Do you want to append any other string messages to them?